dev/gen_version.sh

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+export LC_ALL=C
+COMPILE_DATE=`date`
+
+if test -x `which git`
+then
+    GIT_DESCRIBE=`git describe --tags --first-parent --dirty | sed 's/^V//'`
+    GIT_BRANCH=`git branch -a | sed -ne '/^\* /s/^\* \(.*\)/\1/p'`
+    GIT_HASH=`git rev-parse HEAD`
+    echo "type version = { describe : string; branch : string; hash : string } ;;"
+    echo "let version = { describe = \"${GIT_DESCRIBE}\"; branch = \"${GIT_BRANCH}\"; hash = \"${GIT_HASH}\" } ;;"
+    echo "let compile_date = \"${COMPILE_DATE}\" ;;"
+else
+    echo "let version = \"unknown\" ;;"
```

Comment:
This is fragile, isn't it? It will be often the case that Coq is not compiled from inside a git repository, e.g. when people download the archive associated to a release. In that case, the default value was supposed to be the `coq_version` variable from the `configure.ml` file.

Reply:
Moreover, I would like at least the possibility to distribute Coq with a pre-generated `version/coqversion.ml` for the case where we are not in a git repository (e.g. OPAM). It seems that your solution is going to erase the pre-generated file and replace it with something saying "version unknown". That said, the `compile_date` should still get erased so the solution probably involves some refactoring.

Reply:
Indeed, this is a change of behavior compared to my original PR. I think we should stick with a default value being manually provided.

Reply:
I used unknown as if git is not present the information on `coq_version` tends to be useless. Anyways I am happy to make that `let version = Coq_config.coq_version`.

Reply:
Why do you think it is useless? When the sources are the archive associated to a release, then `coq_version` is *exactly* the right version string.

And what about my second comment (ability to ship a pre-generated file)? This is also something that I liked about Maxime's initial version.